### PR TITLE
 week02_submission

### PR DESCRIPTION
**Pull request by rubayamim381**

### CODE REVIEW

File: week02/Longest Substring with At Least K Repeating Characters.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Approach 1: Recursion
+// TC:  O(26 * n) = O(n), MC: O(1)
+
+class Solution {
+public:
+
+    int max_len = 0;
```
Comment (Sadman007) [inline week02/Longest Substring with At Least K Repeating Characters.cpp:7]: It must be a local variable.

File: week02/decode string.cpp [hunk 2]
```diff
@@ -0,0 +1,53 @@
+//tc:O(n), MC:O(n)
+class Solution {
+public:
+    string decodeString(string s) {
+
+        stack <pair<string, int>> DecodedStr_stk;
+        DecodedStr_stk.push({"", 1});
```
Comment (Sadman007) [inline week02/decode string.cpp:7]:
```suggestion
        decoded_str_stk.push({"", 1});

```

File: week02/group anagrams.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+            long long H = 1;
+
+            for(auto ch : word)
+            {
+                H *= base + ch;
+                if( H >= mod ) H %= mod;
+            }
+
+            map_strings[H].push_back(word);
+        }
```
Comment (Sadman007) [inline week02/group anagrams.cpp:29]: Keep it in a separate function.

File: week02/largest number.cpp [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        //sorted according to each character's Unicode code point value
+        //a=The first element for comparison.
+        //b=The second element for comparison.
+
+        sort(nums.begin(),nums.end(),[&](const int& a,const int& b){
+
+            string s1 = to_string(a) + to_string(b);
+            string s2 = to_string(b) + to_string(a);
+            return s1 > s2;
+        });
```
Comment (Sadman007) [inline week02/largest number.cpp:17]: This is a C++ lambda expression. Are you sure you understand how lambda works? I would recommend not taking a solution from any discussion forum and using it without understanding properly.

**Review by Sadman007 — APPROVED** [review]
Hi @rubayamim381 

Work on the updates and later feel free to merge this PR. thanks.

File: week02/largest number.cpp (hunk 4 above)
Comment (rubayamim381) [reply week02/largest number.cpp:17]: To be honest, I did not know much about the lambda function before this solution. I first learnt about it for this particular solution, then I use it here. But I do not know much more about this function. I understood the problem, but, I could not figure out how to solve it in a optimal way, that's why, Yes, I took the concept of this solution from google and leetcode discussion. Then I learnt what I did not know, then I used it.

**Pull request merged**